src/fstext/fstext-utils-inl.h

```diff
@@ -681,6 +677,7 @@ VectorFst<Arc>* MakeLoopFst(const vector<const ExpandedFst<Arc> *> &fsts) {
   unordered_map<const ExpandedFst<Arc> *, Arc> cache;
 
   for (Label i = 0; i < static_cast<Label>(fsts.size()); i++) {
+    // TODO(galv): I feel like this won't work with my unique_ptr usage. Call .get()?
     const ExpandedFst<Arc> *fst = fsts[i];
```

Comment:
maybe if you made this a reference and dereferenced the unique_ptr, it would work?

Reply:
.get() works fine. The problem with * is that sometimes the std::unique_ptr is allowed to point to NULL (for reasons I haven't taken the time to understand fully)

Reply:
In general, it is! (I do not resolve the ambiguity of limiting in your use of "allowed", in general, or in your algorithm). You can assign it a nullptr (this deletes the previous object), you can steal a pointer from it via .release etc. The default constructor constructs one without a pointer, too. `.get()` is a very common pattern.

Reply:
I am not sure what is the source of your doubt exactly, but your pattern looks absolutely correct and totally idiomatic to me. This won't compile:
```c++
  const ExpandedFst<Arc> *fst = fsts[i];  // No conversion exists from unique_ptr<T> to const ExpandedFst<Arc> *
```
because unique_ptr has no defined conversion operator to the pointer type. This is entirely idiomatic:
```c++
  const ExpandedFst<Arc> *fst = fsts[i].get();  // Well-defined.
  if (fst == nullptr) . . .
```
You would worry about nullity of the pointer if you attempted to dereference it, e. g.
```c++
  const ExpandedFst<Arc> &fst = *fsts[i];  // Possible UB.
  ExpandedFst<Arc> fst = *fsts[i];  // Possible UB.
```
which is UB if the pointer is null. But the `.get()` has a defined behavior for a null unique_ptr. Another thing that the `unique_ptr` has is a comparison with the `nullptr`:
```c++
  if (fsts[i] == nullptr) . . . // Well-defined.
```
but not with other pointer types; `nullptr` is the only value of the type `nullptr_t`, and the operators `==` and `!=` are defined only for this type.